pallet/doughnut/src/lib.rs

```diff
@@ -0,0 +1,557 @@
+				let Ok(futurepass) = <T as Config>::FuturepassLookup::lookup(fee_payer_address.clone().into()) else {
+					log!(error,"⛔️ failed to retrieve futurepass address for the address: {:?}", fee_payer_address);
+					return None
+				};
+				fee_payer_address = futurepass.into();
+			}
+
+			let sender_address = T::AccountId::from(*info);
+
+			// Pre dispatch
+			// Create the validation instances for this extrinsic
+			let validations_fee_payer: DoughnutFeePayerValidations<T> = (
+				ChargeTransactionPayment::<T>::from((*tip).into()),
+			);
+			let validations_sender: DoughnutSenderValidations<T> = (
+				CheckNonZeroSender::new(),
+				CheckNonce::from(nonce.clone().into()),
+				CheckWeight::new(),
+			);
+
+			let _pre_sender = SignedExtension::pre_dispatch(validations_sender, &sender_address, &(**inner_call).clone().into(), dispatch_info, len).ok()?;
+			let pre_issuer = SignedExtension::pre_dispatch(validations_fee_payer, &fee_payer_address, &(**inner_call).clone().into(), dispatch_info, len).ok()?;
+
+			// Dispatch the outer call. i.e Doughnut::transact() with None as the origin
+			let res = call.dispatch(frame_system::RawOrigin::None.into());
+			let post_info = match res {
+				Ok(info) => info,
+				Err(err) => err.post_info,
+			};
+
+			// post dispatch
+			<DoughnutFeePayerValidations<T> as SignedExtension>::post_dispatch(
+				Some(pre_issuer),
+				dispatch_info,
+				&post_info.into(),
+				len,
+				&res.map(|_| ()).map_err(|e| e.error),
+			).ok()?;
+
+			return Some(res)
+		}
+		None
+	}
+}
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use seed_pallet_common::ExtrinsicChecker;
+
+	const STORAGE_VERSION: StorageVersion = StorageVersion::new(0);
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub (super) trait Store)]
+	#[pallet::storage_version(STORAGE_VERSION)]
+	pub struct Pallet<T>(PhantomData<T>);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config
+	where
+		<Self as frame_system::Config>::AccountId: From<H160>,
+	{
+		/// The overarching event type
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+		/// The overarching call type.
+		type RuntimeCall: Parameter
+			+ Dispatchable<RuntimeOrigin = Self::RuntimeOrigin>
+			+ GetDispatchInfo
+			+ From<frame_system::Call<Self>>
+			+ IsSubType<Call<Self>>
+			+ GetCallMetadata;
+		/// Inner call validator
+		type CallValidator: ExtrinsicChecker<
+			Call = <Self as Config>::RuntimeCall,
+			Extra = Topping,
+			Result = DispatchResult,
+		>;
+		/// A lookup to get futurepass account id for a futurepass holder.
+		type FuturepassLookup: StaticLookup<Source = H160, Target = H160>
+			+ ExtrinsicChecker<
+				Call = <Self as Config>::RuntimeCall,
+				Extra = (),
+				Result = DispatchResult,
+			>;
+		/// Weight information for the extrinsic call in this module.
+		type WeightInfo: WeightInfo;
+	}
+
+	/// Map from collection to its public minting information
```

Comment:
Oops, I think this was me, but this comment is wrong 
```suggestion
	/// Map doughnut hash to blocked status
```

Reply:
[62e90bf](https://github.com/futureversecom/trn-seed/pull/760/commits/62e90bf935de42565c9a51b52f1e6c1fd2f680a8)